CMakeLists.txt

```diff
@@ -57,6 +59,10 @@ include(cmake/third_party/snappy.cmake)
 include(cmake/third_party/zlib.cmake)
 include(cmake/third_party/sodium.cmake)
 
+if(NOT ENABLE_SHARED AND NOT ENABLE_STATIC)
+    message(FATAL_ERROR "Both ENABLE_SHARED & ENABLE_STATIC are disabled. Need to enable at least one build flavour.")
+endif()
```

Comment:
I'd expect at least one of the 2 would have a default of "yes", that is, even if not explicitly specified there won't be a case both are disabled? 

Btw, I don't think we have the set of "disable-xxx" config options in the CMake world. I guess that means semantically if "enable-xxx" is not specified the corresponding option is disabled?

Reply:
True, though this check was to mainly test against user input errors, as an incorrect configure command can end up in this state. For example, the user can still do something like the following `cmake -DENABLE_SHARED=0 -DENABLE_STATIC=0 ../.`. Though that seems like a weird this to type, more probable permutation that results in the previous example could be:
```
cmake -DENABLE_STATIC=0 ../.
...
cmake -DENABLE_SHARED=0 ../.
```
This test was just to prevent user errors. In most cases, we should have at least one of the 2 sets.

On the later, correct. Semantically the disable equivalent would be `-DENABLE_SHARED=0`. When I initially added them a while ago, I wanted to keep the set of configuration options simple to avoid having to do messy edge case checks i.e. needing to resolve `-DENABLE_SHARED=0 -DDISABLE_SHARED=1` (it probably easier in autoconf since the command options can be parsed sequentially, but we can't do that in CMake)

Reply:
Ah, forgot we could set a "0" value. All make sense now. Cheers